### PR TITLE
Fr/feature/rest mdx components

### PR DESCRIPTION
**Pull request by RavenInd**
Create MD components:
- Table
- HorizontalRule
- Image
- Link (navigation and external)
- Headings lvl 1,4,5
- UnorderedList

### CODE REVIEW

File: docs/src/pages/[...docPath].tsx [hunk 1]
```diff
@@ -64,6 +77,16 @@ interface IDocPage {
 }
 
 const DocPage: NextPage<IDocPage> = ({ source, frontMatter }): JSX.Element => {
+  const router = useRouter();
+
+  useEffect(() => {
+    const documentationView = document?.getElementById(
+      "documentation-view"
```
Comment (usulpro) [inline docs/src/pages/[...docPath].tsx]: why we can't use React useRef API instead? For that we would need to move this logic to DocumentLayout but for me it looks reasonable (e.g. if we will add another page with the same layout we wouldn't need to duplicate this logic)

File: docs/src/content/getting-started/index.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ Welcome to Hardhat v2.0.8
 
 Let’s create the sample project and go through these steps to try out the sample task and compile, test and deploy the sample contract.
 
-The sample project will ask you to install `hardhat-waffle` and `hardhat-ethers`, which makes Hardhat compatible with tests built with Waffle. You can learn more about it [in this guide](../guides/waffle-testing.md).
+The sample project will ask you to install `hardhat-waffle` and `hardhat-ethers`, which makes Hardhat compatible with tests built with Waffle. You can learn more about it [in this guide](/guides/waffle-testing.md).
```
Comment (usulpro) [inline docs/src/content/getting-started/index.md]: What reason for switching from relative to absulute path here? we have a requirement to keep links workable on Github as well - don't we break it now?

File: docs/src/pages/[...docPath].tsx (hunk 1 above)
Comment (usulpro) [reply docs/src/pages/[...docPath].tsx]: @RavenInd

File: docs/src/content/getting-started/index.md (hunk 2 above)
Comment (usulpro) [reply docs/src/content/getting-started/index.md]: @RavenInd

File: docs/src/pages/[...docPath].tsx (hunk 1 above)
Comment (RavenInd) [reply docs/src/pages/[...docPath].tsx]: corrected

File: docs/src/content/getting-started/index.md (hunk 2 above)
Comment (RavenInd) [reply docs/src/content/getting-started/index.md]: it was for experiment purposes. 
corrected

**Review by usulpro — APPROVED** [review]

**Comment by usulpro** [comment]
Closing in favor of #35

**Pull request closed without merging**